### DISCUSSION TITLE
Normalize use of whitespace

### DISCUSSION THREAD

**Discussion by scandariato** [megastep/makeself#299] [General]
The various files making up makeself have mixed tabs and spaces for indentation, as well as random trailing whitespace.

It would be useful to convert all tabs to spaces (or vice versa) - pick one and document it, please!

Also, remove all trailing whitespace.

**Comment by realtime-neil** [comment]
@megastep , when (and if) you're okay with sacrificing `git blame` clarity for source code uniformity, I would like to propose reformatting the source with the following tool:

https://github.com/mvdan/sh

I have used `shfmt` for a little over two years with good results. I usually invoke it like this...

```
shfmt -ln posix -i 4 -ci -bn
```
...which parses the `posix` shell language variant, forces indentation to use four spaces, indents switch cases, and allows binary operators to begin lines.

Here's what the options for `v3.0.0-alpha1` look like:
```
$ shfmt -version && shfmt -help
v3.0.0-alpha1
usage: shfmt [flags] [path ...]

If no arguments are given, standard input will be used. If a given path
is a directory, it will be recursively searched for shell files - both
by filename extension and by shebang.

  -version  show version and exit

  -l        list files whose formatting differs from shfmt's
  -w        write result to file instead of stdout
  -d        error with a diff when the formatting differs
  -s        simplify the code

Parser options:

  -ln str   language variant to parse (bash/posix/mksh, default "bash")
  -p        shorthand for -ln=posix

Printer options:

  -i uint   indent: 0 for tabs (default), >0 for number of spaces
  -bn       binary ops like && and | may start a line
  -ci       switch cases will be indented
  -sr       redirect operators will be followed by a space
  -kp       keep column alignment paddings
  -mn       minify program to reduce its size (implies -s)

Utilities:

  -f        recursively find all shell files and print the paths
  -tojson   print syntax tree to stdout as a typed JSON
```
